src/EnergyPlus/ConvectionCoefficients.cc

```diff
@@ -4079,6 +4080,115 @@ void CalcISO15099WindowIntConvCoeff(EnergyPlusData &state,
+                auto itnext = std::next(it);
+                if (itnext == std::end(vertices)) {
+                    itnext = std::begin(vertices);
+                }
+
+                auto curVertex = *it;
+                auto nextVertex = *itnext;
+                if (uniqueRoofVertices.size() == 0) {
+                    uniqueRoofVertices.emplace_back(curVertex);
+                } else {
+                    auto it2 = std::find_if(uniqueRoofVertices.begin(), uniqueRoofVertices.end(), [&curVertex](const auto &unqV) {
+                        return SurfaceGeometry::isAlmostEqual3dPt(curVertex, unqV);
+                    });
+                    if (it2 == std::end(uniqueRoofVertices)) {
+                        uniqueRoofVertices.emplace_back(curVertex);
+                    }
+                }
+
+                SurfaceGeometry::EdgeOfSurf thisEdge;
+                thisEdge.start = curVertex;
+                thisEdge.end = nextVertex;
+                thisEdge.count = 1;
+
+                // Uses the custom operator== that uses isAlmostEqual3dPt internally and doesn't care about order of the start/end
+                auto itEdge = std::find(uniqEdgeOfSurfs.begin(), uniqEdgeOfSurfs.end(), thisEdge);
+                if (itEdge == uniqEdgeOfSurfs.end()) {
+                    uniqEdgeOfSurfs.emplace_back(std::move(thisEdge));
+                } else {
+                    ++(itEdge->count);
+                }
+            }
+        }
+    }
+
+    RoofGeo.Height /= RoofGeo.Area;
+    RoofGeo.Tilt /= RoofGeo.Area;
+    RoofGeo.Azimuth /= RoofGeo.Area;
+
+    // Remove the ones that are already used twice
+    uniqEdgeOfSurfs.erase(std::remove_if(uniqEdgeOfSurfs.begin(), uniqEdgeOfSurfs.end(), [](const auto &edge) -> bool { return edge.count == 2; }),
+                          uniqEdgeOfSurfs.end());
+
+    // Intersect with unique vertices as much as needed
+    bool insertedVertext = true;
+    while (insertedVertext) {
+        // Use an index-based loop because I may emplace_back inside the loop, and that invalidates the iterators
+        for (size_t i = 0; i < uniqEdgeOfSurfs.size(); ++i) {
+
+            auto &edge = uniqEdgeOfSurfs[i];
+
+            insertedVertext = false;
+
+            // now go through all the vertices and see if they are colinear with start and end vertices
+            for (const auto &testVertex : uniqueRoofVertices) {
+                if (edge.containsPoints(testVertex)) {
+                    SurfaceGeometry::EdgeOfSurf newEdgeOfSurface;
+                    newEdgeOfSurface.start = testVertex;
+                    newEdgeOfSurface.end = edge.end;
+                    edge.end = testVertex;
+                    uniqEdgeOfSurfs.emplace_back(std::move(newEdgeOfSurface));
+                    break;
+                }
+            }
+            // Break out of the loop on edges, and start again at the while
+            if (insertedVertext) {
+                break;
+            }
+        }
+    }
```

Comment:
Step 2: try to break EdgeOfSurfs into several if a vertex is found to actually be on it. Do that until no insertion is possible.